### PR TITLE
Adding Build Step To Support Optional Auto-Release

### PR DESCRIPTION
**Pull request by scbedd**
The end result is that the build get's tagged with `RELEASE_CANDIDATE` if a queue time variable is set.

Release triggers can then take over the rest.

@lmazuel this is the last piece.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by scbedd** [comment]
@Azure/azure-sdk-eng

File: .azure-pipelines/client.yml [hunk 1]
```diff
@@ -34,6 +36,10 @@ jobs:
           twine check $(Build.ArtifactStagingDirectory)/*
         displayName: 'Verify Readme'
 
+      - script: 'echo "##vso[build.addbuildtag]$(ReleaseTag)"'
```
Comment (weshaggard) [inline .azure-pipelines/client.yml]: Why does what with this special build variable? Are there any documents you can point me to?

Comment (scbedd) [reply .azure-pipelines/client.yml]: This `tags` the build. AKA  a `DevOps` tag. Found [here](https://github.com/microsoft/azure-pipelines-tasks/blob/master/docs/authoring/commands.md#build-logging-commands)

**Pull request merged**